Source/CarthageKit/Xcode.swift

```diff
@@ -355,6 +355,40 @@ private func mergeExecutables(_ executableURLs: [URL], _ outputURL: URL) -> Sign
 		.then(SignalProducer<(), CarthageError>.empty)
 }
 
+private func mergeSwiftHeaderFiles(_ simulatorExecutableURL: URL,
+								   _ deviceExecutableURL: URL,
+								   _ executableOutputURL: URL) -> SignalProducer<(), CarthageError> {
+	precondition(simulatorExecutableURL.isFileURL)
+	precondition(deviceExecutableURL.isFileURL)
+	precondition(executableOutputURL.isFileURL)
+	
+	let conditionalPrefix = "#if TARGET_OS_SIMULATOR\n"
+	let conditionalElse = "\n#else\n"
+	let conditionalSuffix = "\n#endif"
+	
+	let conditionalPrefixContents = conditionalPrefix.data(using: .utf8)!
+	let conditionalElseContents = conditionalElse.data(using: .utf8)!
+	let conditionalSuffixContents = conditionalSuffix.data(using: .utf8)!
+	
+	guard let simulatorHeaderURL = simulatorExecutableURL.deletingLastPathComponent().swiftHeaderURL() else { return .empty }
+	guard let simulatorHeaderContents = FileManager.default.contents(atPath: simulatorHeaderURL.path) else { return .empty }
+	guard let deviceHeaderURL = deviceExecutableURL.deletingLastPathComponent().swiftHeaderURL() else { return .empty }
+	guard let deviceHeaderContents = FileManager.default.contents(atPath: deviceHeaderURL.path) else { return .empty }
+	guard let outputURL = executableOutputURL.deletingLastPathComponent().swiftHeaderURL() else { return .empty }
+	
+	var fileContents = Data()
+	
+	fileContents.append(conditionalPrefixContents)
+	fileContents.append(simulatorHeaderContents)
+	fileContents.append(conditionalElseContents)
+	fileContents.append(deviceHeaderContents)
+	fileContents.append(conditionalSuffixContents)
+	
+	FileManager.default.createFile(atPath: outputURL.path, contents: fileContents, attributes: nil)
```

Comment:
In the case where overwriting this file doesn't succeed, `FileManager.createFile` will output `false` — we should check for that — erroring out, rather than letting a potential non-overwrite of that header go unaddressed:

```swift
        switch FileManager.default.createFile(atPath: outputURL.path, contents: fileContents) {
        case false: return .init(error: .writeFailed(outputURL, nil))
        case true: return .empty
        }
```

Reply:
Good catch, should be resolved now.

Reply:
Wow, fantastic, very fast — thank you!